src/main/java/uk/gov/ons/census/casesvc/service/CaseService.java

```diff
@@ -60,6 +70,35 @@ public Case saveCase(CreateCaseSample createCaseSample) {
     return caze;
   }
 
+  public Case buildCCSCase(
+      String caseId, SampleUnitDTO sampleUnit, String actionPlanId, String collectionExerciseId) {
+    int caseRef = getUniqueCaseRef();
+
+    Case caze = mapperFacade.map(sampleUnit, Case.class);
+    caze.setCaseRef(caseRef);
+    caze.setCaseType(sampleUnit.getAddressType());
+    caze.setCaseId(UUID.fromString(caseId));
+    caze.setActionPlanId(actionPlanId);
+    caze.setCollectionExerciseId(collectionExerciseId);
+    caze.setState(CaseState.ACTIONABLE);
+    caze.setCreatedDateTime(OffsetDateTime.now());
+    caze.setCcsCase(true);
+
+    return caze;
+  }
+
+  public Case saveCCSCaseWithUacQidLink(Case caze, UacQidLink uacQidLink) {
+    uacQidLinkRepository.saveAndFlush(uacQidLink);
+
+    caze.setUacQidLinks(new LinkedList<>(Collections.singletonList(uacQidLink)));
+    caseRepository.saveAndFlush(caze);
+
+    uacQidLink.setCaze(caze);
```

Comment:
This is perfectly adequate for JPA/Hibernate to link the case and the UAC-QID

Reply:
done